runtime/runtime-params-estimator/src/estimator_context.rs

```diff
@@ -133,6 +137,48 @@ impl<'c> EstimatorContext<'c> {
             migration_flags: MigrationFlags::default(),
         }
     }
+
+    #[cfg(feature = "protocol_feature_flat_state")]
+    fn create_flat_state_factory(store: Store) -> FlatStateFactory {
+        // function-level `use` statements here to avoid `unused import` warning
+        // when flat state feature is disabled
+        use near_primitives::types::BlockHeight;
+        use near_store::flat_state::{
+            store_helper, BlockInfo, ChainAccessForFlatStorage, FlatStorageState,
+        };
+        use std::collections::HashSet;
+
+        struct ChainAccess;
+
+        impl ChainAccessForFlatStorage for ChainAccess {
+            fn get_block_info(&self, block_hash: &CryptoHash) -> BlockInfo {
+                BlockInfo { hash: block_hash.clone(), prev_hash: Default::default(), height: 0 }
+            }
+
+            fn get_block_hashes_at_height(
+                &self,
+                _block_height: BlockHeight,
+            ) -> HashSet<CryptoHash> {
+                unimplemented!()
+            }
+        }
+
+        let shard_id = 0;
+        let block_height = 0;
+        let mut store_update = store.store_update();
+        store_helper::set_flat_head(&mut store_update, shard_id, &FLAT_STATE_HEAD);
```

Comment:
some comments somewhere about how we "fake" flat state for the estimator would also be nice

(We will eventually need to do it properly, I assume, as we want the estimator to estimate real storage costs. But that's for later and can potentially be done even after stabilization of flat storage.)

Reply:
good point, I've added some comments